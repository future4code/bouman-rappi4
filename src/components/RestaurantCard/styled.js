import styled from "styled-components";

export const CardContainer = styled.div `
    width: 90%;
    min-height: 200px;
    margin: 10px;
    display: flex;
    flex-direction: column;
    justify-content: flex-start;
    align-items: center;
    background-color: #ffffff;
    border-radius: 5px;
    border: 0.5px solid #bdbdbd;
    :hover {
        cursor: pointer;
    }
`

export const RestaurantName = styled.div `
    width: 100%;
    height: 10px;
    color: #E86E5A;
    padding: 10px;
    font-weight: bolder;
`

export const ShippingAndDeliveryContainer = styled.div `
    width: 100%;
    padding: 10px;
    display: flex;
    justify-content: space-between;
    align-items: center;
    color: #bdbdbd;
    font-weight: bolder;
`

export const CardImg = styled.img `
    width: 100%;
    height: 20vh;
    border-radius: 5px;
`