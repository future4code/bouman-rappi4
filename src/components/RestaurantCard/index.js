import React from 'react';
import { CardContainer, CardImg, RestaurantName, ShippingAndDeliveryContainer } from './styled'

function RestaurantCard(props) {
    return (
        <CardContainer key={props.key} onClick={props.onClick}>
            <CardImg src={props.img}/>
            <RestaurantName>{props.name}</RestaurantName>
            <ShippingAndDeliveryContainer>
                <div>{props.deliveryTime} min </div>
                <div>Frete: R${props.price},00</div>
            </ShippingAndDeliveryContainer> 
        </CardContainer>
    )
}

export default RestaurantCard;
