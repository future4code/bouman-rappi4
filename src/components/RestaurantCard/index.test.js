import React from 'react'
import renderer from "react-test-renderer";
import  RestaurantCard  from './index';

describe ('teste componente RestaurantCard', () => {
    it('renders correctly', () => {
        const tree = renderer
            .create(<RestaurantCard></RestaurantCard>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})