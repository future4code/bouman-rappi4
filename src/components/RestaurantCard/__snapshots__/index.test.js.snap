// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`teste componente RestaurantCard renders correctly 1`] = `
<div
  className="sc-bdVaJa czOKIc"
>
  <img
    className="sc-bxivhb iechTi"
  />
  <div
    className="sc-bwzfXH fSIllJ"
  />
  <div
    className="sc-htpNat dksbWW"
  >
    <div>
       min 
    </div>
    <div>
      Frete: R$
      ,00
    </div>
  </div>
</div>
`;
